### PR TITLE
openstack-ardana-gating: full tempest and entry-scale-kvm

### PR DESCRIPTION
**Pull request by stefannica**
This change replace the `std-min` and `std-3cp` input models
currently employed by the Ardana gating jobs with an
entry-scale-kvm input model and replaces the 'ci' tempest
filter with a full set of tempest filters.
    
With the recent cleanup of Ardana tempest filters [1] [2] and the
efforts to stabilize the CI, we can finally have a full tempest
coverage in the gate. The `entry-scale-kvm` input model is
also better suited for a gating job, because it is closer to
what customers use.
    
This change also updates all other jobs to reflect the new
tempest filters and run only those that are applicable [3].

These are the main differences between the `standard` and
`entry-scale-kvm` input model scenarios:
 - there is no dedicated CLM (CONF) network in `entry-scale-kvm`.
 It is part of the MANAGEMENT network
 - there is only one neutron VLAN tenant network in `entry-scale-kvm`,
 which is not associated with the management network. This prevents
 problems with Neutron having to reconfigure the interface associated
 with the management network
 - there is no dedicated SWIFT network in `entry-scale-kvm`, swift services
 use the same management network as the rest of the services
 - last but not least, there is an important difference in the interface models: 
 `entry-scale-kvm` only uses two NICs, whereas the `standard` one basically
 has one interface per network group per server. Another important difference
 is the fact that `entry-scale-kvm` does not exercise interface bonding.

[1] https://gerrit.prv.suse.net/c/ardana/tempest-ansible/+/5904
[2] https://gerrit.prv.suse.net/c/ardana/tempest-ansible/+/5986
[3] https://confluence.suse.de/display/qualityperformance/Tempest+Test+Coverage+on+Ardana

### CODE REVIEW

**Comment by stefannica** [comment]
I started a manual job for entry-scale-kvm on cloud9 with all the relevant Gerrit changes included: https://ci.suse.de/blue/organizations/jenkins/cloud-ardana9-job-entry-scale-kvm-x86_64/detail/cloud-ardana9-job-entry-scale-kvm-x86_64/168/pipeline

**Comment by rtamalin** [comment]
@stefannica What is different between an entry-scale-kvm and the std-min models? The original ardana-input-model static standard, and thus the derived std-min model, is intended to deploy as much of the possible Ardana functionality as possible, giving as much coverage as possible in terms of deployed service components...

The dynamically generated versions used in the CI already disable some bits, e.g. the Swift EC ringe is commented out, but I'm just wondering if we are losing anything else by switching to the entry-scale-kvm base model?

File: jenkins/ci.suse.de/cloud-ardana8.yaml [hunk 1]
```diff
@@ -131,20 +131,42 @@
...
+    ardana_env: cloud-ardana-ci-slot
+    cloudsource: stagingcloud8
+    scenario_name: entry-scale-kvm
+    clm_model: standalone
+    controllers: '3'
+    sles_computes: '2'
+    rhel_computes: '0'
+    ses_enabled: true
+    ses_rgw_enabled: false
+    tempest_filter_list: "\
```
Comment (rtamalin) [inline jenkins/ci.suse.de/cloud-ardana8.yaml]: Why use ugly backslash based continuation here when you could use `>-`?

File: jenkins/ci.suse.de/cloud-ardana8.yaml [hunk 2]
```diff
@@ -131,20 +131,42 @@
...
     clm_model: standalone
     controllers: '3'
-    sles_computes: '1'
+    sles_computes: '2'
+    rhel_computes: '0'
     ses_enabled: true
     ses_rgw_enabled: false
-    triggers:
-     - timed: 'H H * * *'
+    tempest_filter_list: "\
```
Comment (rtamalin) [inline jenkins/ci.suse.de/cloud-ardana8.yaml]: As above... Why not use `>-`?

File: jenkins/ci.suse.de/cloud-ardana8.yaml [hunk 3]
```diff
@@ -176,7 +198,9 @@
     cloudsource: GM8+up
     updates_test_enabled: false
     scenario_name: entry-scale-kvm
-    tempest_filter_list: "smoke,full"
+    tempest_filter_list: "\
```
Comment (rtamalin) [inline jenkins/ci.suse.de/cloud-ardana8.yaml]: As above... Why not use `>-`?

File: jenkins/ci.suse.de/cloud-ardana8.yaml [hunk 4]
```diff
@@ -214,15 +238,8 @@
     rhel_computes: '0'
     ses_enabled: true
     tempest_filter_list: "\
```
Comment (rtamalin) [inline jenkins/ci.suse.de/cloud-ardana8.yaml]: And yet again - why not use `>-`?

File: jenkins/ci.suse.de/cloud-ardana9.yaml [hunk 5]
```diff
@@ -122,14 +122,34 @@
...
     rhel_computes: '0'
-    tempest_filter_list: 'ci'
     ses_enabled: true
     ses_rgw_enabled: false
-    qa_test_list: "\
-      iverify,cinder,heat,magnum,neutron,nova-attach,nova_volume,nova_server,\
-      nova_services,nova_flavor,nova_image,tempest_cleanup,nova_guest_image"
-    triggers:
-     - timed: 'H H * * *'
+    tempest_filter_list: "\
```
Comment (rtamalin) [inline jenkins/ci.suse.de/cloud-ardana9.yaml]: And another case... Why not use `>-`?

File: jenkins/ci.suse.de/cloud-ardana9.yaml [hunk 6]
```diff
@@ -122,14 +122,34 @@
...
+    update_after_deploy: true
+    update_to_cloudsource: stagingcloud9
+    scenario_name: entry-scale-kvm
+    clm_model: standalone
+    controllers: '3'
+    sles_computes: '2'
+    rhel_computes: '0'
+    ses_enabled: true
+    ses_rgw_enabled: false
+    tempest_filter_list: "\
```
Comment (rtamalin) [inline jenkins/ci.suse.de/cloud-ardana9.yaml]: Same again... Why not use `>-`

File: jenkins/ci.suse.de/cloud-ardana8.yaml (hunk 1 above)
Comment (stefannica) [reply jenkins/ci.suse.de/cloud-ardana8.yaml]: You're right, it hasn't occurred to me that we can use `>-`. Done

File: jenkins/ci.suse.de/cloud-ardana8.yaml (hunk 3 above)
Comment (stefannica) [reply jenkins/ci.suse.de/cloud-ardana8.yaml]: Done

File: jenkins/ci.suse.de/cloud-ardana9.yaml (hunk 5 above)
Comment (stefannica) [reply jenkins/ci.suse.de/cloud-ardana9.yaml]: Done here too

File: jenkins/ci.suse.de/cloud-ardana9.yaml (hunk 6 above)
Comment (stefannica) [reply jenkins/ci.suse.de/cloud-ardana9.yaml]: Done

**Comment by stefannica** [comment]
> @stefannica What is different between an entry-scale-kvm and the std-min models? The original ardana-input-model static standard, and thus the derived std-min model, is intended to deploy as much of the possible Ardana functionality as possible, giving as much coverage as possible in terms of deployed service components...
> 
> The dynamically generated versions used in the CI already disable some bits, e.g. the Swift EC ringe is commented out, but I'm just wondering if we are losing anything else by switching to the entry-scale-kvm base model?

I'll give this a closer look, but to put your mind at ease right away, the QA team is mostly using the `entry-scale-kvm` generated input model to run validation testing and very rarely the `standard` ones.

**Comment by stefannica** [comment]
> I started a manual job for entry-scale-kvm on cloud9 with all the relevant Gerrit changes included: https://ci.suse.de/blue/organizations/jenkins/cloud-ardana9-job-entry-scale-kvm-x86_64/detail/cloud-ardana9-job-entry-scale-kvm-x86_64/169/pipeline

I started two temporarily rigged gating jobs to test this:
 - cloud8: https://ci.suse.de/blue/organizations/jenkins/cloud-ardana8-gating/detail/cloud-ardana8-gating/181/pipeline/28
 - cloud9: https://ci.suse.de/blue/organizations/jenkins/cloud-ardana9-gating/detail/cloud-ardana9-gating/456/pipeline

File: jenkins/ci.suse.de/cloud-ardana8.yaml [hunk 7]
```diff
@@ -131,20 +131,42 @@
...
     clm_model: standalone
     controllers: '3'
-    sles_computes: '1'
+    sles_computes: '2'
+    rhel_computes: '0'
     ses_enabled: true
     ses_rgw_enabled: false
-    triggers:
-     - timed: 'H H * * *'
+    tempest_filter_list: >-
```
Comment (rtamalin) [inline jenkins/ci.suse.de/cloud-ardana8.yaml]: Something is not right here as the >- should be stripping off the trailing whitespace, and the leading whitespace should be being ignored... However the jobs shows a failure for running the designate filter (https://ci.suse.de/blue/organizations/jenkins/cloud-ardana9-job-entry-scale-kvm-x86_64/detail/cloud-ardana9-job-entry-scale-kvm-x86_64/175/pipeline/252#step-253-log-8) as follows:
[2019-05-01T07:11:05.537Z] Running: ansible-playbook run-tempest.yml -e @input.yml -e tempest_run_filter= designate
[2019-05-01T07:11:06.360Z] ERROR! the playbook: designate could not be found

That error suggests that the whitespace before the designate at the start of the above line got included in the value of the filters list... So the previous pattern using a quote string with \ continuation may have been correct... My bad...

Comment (stefannica) [reply jenkins/ci.suse.de/cloud-ardana8.yaml]: No worries, I reverted the change :)

**Comment by stefannica** [comment]
> > I started a manual job for entry-scale-kvm on cloud9 with all the relevant Gerrit changes included: https://ci.suse.de/blue/organizations/jenkins/cloud-ardana9-job-entry-scale-kvm-x86_64/detail/cloud-ardana9-job-entry-scale-kvm-x86_64/169/pipeline
> 
> I started two temporarily rigged gating jobs to test this:
> 
> * cloud8: https://ci.suse.de/blue/organizations/jenkins/cloud-ardana8-gating/detail/cloud-ardana8-gating/181/pipeline/28
> * cloud9 (including [all pending Gerrit changes](https://gerrit.prv.suse.net/q/owner:framalho%2540suse.com+status:open)): https://ci.suse.de/blue/organizations/jenkins/cloud-ardana9-gating/detail/cloud-ardana9-gating/460/pipeline

The previous job runs still exhibited tempest failures caused by timeout errors while waiting for a VM to be built. I pushed [this Gerrit patch](https://gerrit.prv.suse.net/c/ardana/tempest-ansible/+/6045) that increases the configured tempest timeout values to match those used by openstack-ansible. Re-running now with that patch included:
 - cloud8: https://ci.suse.de/blue/organizations/jenkins/cloud-ardana8-gating/detail/cloud-ardana8-gating/185/pipeline/29
 - cloud9 (also including [all pending Gerrit changes](https://gerrit.prv.suse.net/q/owner:framalho%2540suse.com+status:open)): https://ci.suse.de/blue/organizations/jenkins/cloud-ardana9-gating/detail/cloud-ardana9-gating/464/pipeline/28

**Comment by stefannica** [comment]
@rtamalin by the way, another issue I noticed is monasca tempest fails consistently after update and reboot, but didn't have time to investigate further.

**Comment by rtamalin** [comment]
> @rtamalin by the way, another issue I noticed is monasca tempest fails consistently after update and reboot, but didn't have time to investigate further.

Something to look into next then ;-)

**Pull request merged**